src/Symfony/Component/HttpClient/CurlHttpClient.php

```diff
@@ -274,6 +274,86 @@ public function request(string $method, string $url, array $options = []): Respo
+                \CURLOPT_INTERFACE => 'bindto',
+                \CURLOPT_TIMEOUT_MS => 'max_duration',
+                \CURLOPT_MAXREDIRS => 'max_redirects',
+                \CURLOPT_PROXY => 'proxy',
+                \CURLOPT_NOPROXY => 'no_proxy',
+                \CURLOPT_SSL_VERIFYPEER => 'verify_peer',
+                \CURLOPT_SSL_VERIFYHOST => 'verify_host',
+                \CURLOPT_CAINFO => 'cafile',
+                \CURLOPT_CAPATH => 'capath',
+                \CURLOPT_SSL_CIPHER_LIST => 'ciphers',
+                \CURLOPT_SSLCERT => 'local_cert',
+                \CURLOPT_SSLKEY => 'local_pk',
+                \CURLOPT_KEYPASSWD => 'passphrase',
+                \CURLOPT_CERTINFO => 'capture_peer_cert_chain',
+                \CURLOPT_USERAGENT => 'normalized_headers',
+                \CURLOPT_REFERER => 'headers',
+                //options used in CurlResponse
+                \CURLOPT_NOPROGRESS => 'on_progress',
+                \CURLOPT_PROGRESSFUNCTION => 'on_progress',
+                \CURLINFO_REDIRECT_COUNT => 'max_redirects',
+            ];
+
+            $curloptsFromResponse = [
+                \CURLOPT_PRIVATE,
+                \CURLOPT_HEADERFUNCTION,
+                \CURLOPT_WRITEFUNCTION,
+                \CURLOPT_VERBOSE,
+                \CURLOPT_STDERR,
+            ];
+
+            $methodOpts = [
+                \CURLOPT_POST,
+                \CURLOPT_PUT,
+                \CURLOPT_CUSTOMREQUEST,
+                \CURLOPT_HTTPGET,
+            ];
+
+            foreach ($options['extra']['curl'] as $opt => $optValue) {
+                if (isset($curloptsToConfig[$opt])) {
+                    $constName = $this->findConstantName($opt) ?? $opt;
+                    throw new InvalidArgumentException(sprintf('Could not override "%s" param, please use "%s" option.', $constName, $curloptsToConfig[$opt]));
+                }
+
+                if (\in_array($opt, $methodOpts)) {
+                    throw new InvalidArgumentException('Could not override method using extra.curl options.');
+                }
+
+                if (\in_array($opt, $curloptsFromResponse)) {
+                    $constName = $this->findConstantName($opt) ?? $opt;
+                    throw new InvalidArgumentException(sprintf('Could not override "%s" option.', $constName));
+                }
+            }
+
+            if (isset($options['extra']['curl'][\CURLOPT_PORT])) {
+                throw new InvalidArgumentException('Could not override URL port using extra.curl options.');
+            }
+
+            if (isset($options['extra']['curl'][\CURLOPT_DNS_USE_GLOBAL_CACHE])) {
+                throw new InvalidArgumentException('Overriding CURLOPT_DNS_USE_GLOBAL_CACHE is impossible.');
+            }
+
+            $curlopts += $options['extra']['curl'];
+            $this->logger && $this->logger->debug(sprintf('There are additional curl options set in request.'));
```

Comment:
Not sure this is useful, I'd suggest removing.

Reply:
removed.